x/mongo/driver/operation.go

```diff
@@ -1553,7 +1553,8 @@ func (op Operation) createReadPref(desc description.SelectedServer, isOpQuery bo
 			doc, _ = bsoncore.AppendDocumentEnd(doc, idx)
 			return doc, nil
 		}
-		doc = bsoncore.AppendStringElement(doc, "mode", "primary")
+
+		return nil, nil
```

Comment:
I understand the logic here after reading the OP_MSG and Server Selection specs, but it's not otherwise obvious why this is correct. Add a comment that describes the expected behavior, specifically that OP_MSG requires never sending read preference "primary" _except_ for topology "single". Also consider noting that we now only support OP_MSG and not OP_QUERY, which has different rules for adding read preference to commands.